### PR TITLE
Codec Factory

### PR DESCRIPTION
**Pull request by qnislavnachev**
A codec factory that is responsible for creating xml and  json codec objects

### CODE REVIEW

**Comment by tsony-tsonev** [comment]
Fix this PR name

**Comment by tsony-tsonev** [comment]
Also add the task description in the description of the PR, and for the rest of the PRs

File: src/task1/AbsFactory.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package task1;
+
+public class AbsFactory implements Factory {
```
Comment (tsony-tsonev) [inline src/task1/AbsFactory.java]: Bad name for both the interface and the class

File: src/task1/AbsFactory.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Override
+    public Codec create(String codec) {
+        if (codec == null) {
+            return null;
+        }
+        if ("JSON".equals(codec)) {
+            return new JSONCodec();
+        }
+        if ("XML".equals(codec)) {
```
Comment (tsony-tsonev) [inline src/task1/AbsFactory.java]: why do we need this type checks, when we are gonna have different factories

Comment (qnislavnachev) [inline src/task1/AbsFactory.java]: I am not sure what do you mean here

Comment (tsony-tsonev) [inline src/task1/AbsFactory.java]: I mean why you need to check for the coded type xml or jeson when you will have to different factories for both of them. Read some more for the factory and you will get what I mean.

File: src/task1/AbsFactory.java (hunk 1 above)
Comment (qnislavnachev) [inline src/task1/AbsFactory.java]: fixed

File: src/task1/AbsFactory.java (hunk 2 above)
Comment (qnislavnachev) [inline src/task1/AbsFactory.java]: fixed

**Comment by tsony-tsonev** [comment]
LGTM

File: src/task1/FactoryProvider.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package task1;
+
+public class FactoryProvider {
+
+    public CodecFactory getFactory(String type) {
```
Comment (qnislavnachev) [inline src/task1/FactoryProvider.java]: fixed

**Pull request merged**